StreamProcessorControllerTest.shouldEnsureMaxSnapshotCount flaky

```java
17:16:14  [ERROR]   StreamProcessorControllerTest.shouldEnsureMaxSnapshotCount:427 
17:16:14  Expected size:<3> but was:<4> in:
17:16:14  <[/tmp/junit2772039577739849483/state-snapshots/4294967424,
17:16:14      /tmp/junit2772039577739849483/state-snapshots/4294967616,
17:16:14      /tmp/junit2772039577739849483/state-snapshots/4294967744,
17:16:14      /tmp/junit2772039577739849483/state-snapshots/4294967808]>
```

We see in the log that the snapshot `4294967424` was deleted.

```java
15:14:34.028 [testProcessor-snapshot-director] [-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Ensure max snapshot count 3, will delete 1 snapshot(s).
15:14:34.091 [testProcessor-snapshot-director] [-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Purged snapshot /tmp/junit2772039577739849483/state-snapshots/4294967424
15:14:34.091 [testProcessor-snapshot-director] [-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Start replicating latest snapshot /tmp/junit2772039577739849483/state-snapshots/4294967808
15:14:34.091 [testProcessor-snapshot-director] [-zb-fs-workers-0] DEBUG io.zeebe.logstr
```

The problem we have is that `Files.delete` is not atomic, see the docs https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#delete-java.nio.file.Path- .
So it can happen that the deletion is successfully done but the main thread does not see this yet.

--- a/logstreams/src/test/java/io/zeebe/logstreams/processor/StreamProcessorControllerTest.java
+++ b/logstreams/src/test/java/io/zeebe/logstreams/processor/StreamProcessorControllerTest.java
@@ -421,11 +421,11 @@ public class StreamProcessorControllerTest {
     }
 
     // then
-    verify(snapshotController, timeout(5000).times(5)).moveValidSnapshot(anyLong());
-    verify(snapshotController, timeout(5000).times(5)).ensureMaxSnapshotCount(3);
+    verify(snapshotController, timeout(5000).atLeast(4)).moveValidSnapshot(anyLong());
+    verify(snapshotController, timeout(5000).atLeast(4)).ensureMaxSnapshotCount(3);
 
+    waitUntil(() -> stateStorage.list().size() == 3);
     assertThat(stateStorage.listByPositionAsc()).hasSize(3);
-    assertThat(stateStorage.list()).hasSize(3);
   }
 
   @Test

